sdk/log/crate/src/lib.rs

```diff
@@ -263,62 +293,13 @@ mod tests {
         }
     }
 
-    /// Generate test cases for unsigned (numeric) types.
-    ///
-    /// The test cases are generated for the given type and buffer size. The
-    /// assert compares that the logger buffer length is less than or equal to
-    /// the maximum length.
-    macro_rules! generate_unsigned_test_case {
-        ( $ty:ident, $max_len:literal, $($size:expr),+ $(,)? ) => {
-            $(
-                let mut logger = Logger::<$size>::default();
-                logger.append($ty::MAX);
-                assert!((*logger).len() <= $max_len);
-            )*
-        };
-    }
-
-    /// Generate test cases for signed (numeric) types.
-    ///
-    /// The test cases are generated for the given type and buffer size. The
-    /// assert compares that the logger buffer length is less than or equal to
-    /// the maximum length.
-    macro_rules! generate_signed_test_case {
-        ( $ty:ident, $max_len:literal, $($size:expr),+ $(,)? ) => {
-            $(
-                let mut logger = Logger::<$size>::default();
-                logger.append($ty::MIN);
-                // Signed types have one extra character for the `-` sign.
-                assert!((*logger).len() <= ($max_len + 1));
-            )*
-        };
-    }
-
-    /// Generate test cases for `str` type.
-    ///
-    /// The test cases are generated for the given value and buffer size. The
-    /// assert compares that the logger buffer length is equal to the minimum
-    /// between the buffer size and the `str` length.
-    macro_rules! generate_str_test_case {
-        ( $str:expr, $($size:expr),+ $(,)? ) => {
-            $(
-                let mut logger = Logger::<$size>::default();
-                logger.append(core::str::from_utf8($str).unwrap());
-                assert_eq!((*logger).len(), core::cmp::min($str.len(), $size));
-            )*
-        };
-    }
-
-    /// Generate a test function for unsigned (numeric) types.
-    ///
-    /// The main objective is to ensure that the logger will not panic when appending
-    /// a value of the type to the buffer.
-    macro_rules! fn_test_unsigned_logger_buffer {
-        ( $test_name:ident, $( ($ty:ident, $max_len:literal) ),+ $(,)? ) => {
-            #[test]
-            fn $test_name() {
-                $(
-                    generate_unsigned_test_case!($ty, $max_len, 1,
+    #[test]
+    fn test_logger_buffer_size_unsigned() {
+        // Test case for an unsigned numeric type.
+        macro_rules! unsigned_test_case {
+            ( $( ($ty:ident, $max_len:literal) ),+ $(,)? ) => {
+                    $(
+                        generate_numeric_test_case!($ty::MIN, ($max_len + 1), 1,
```

Comment:
Since this is unsigned, let's use MAX and remove the + 1, like it was before. Otherwise they're very similar to the signed test cases
```suggestion
                        generate_numeric_test_case!($ty::MAX, $max_len, 1,
```